src/commands.ts

```diff
@@ -428,6 +430,43 @@ function openInExternalEditor(packageNode: PackageNode) {
     }
 }
 
+interface IDawrinCompatibleTarget extends vscode.QuickPickItem {
+    value: DarwinCompatibleTarget;
+    label: string;
+}
+
+async function handleQuickSwitch() {
+    const picker = vscode.window.createQuickPick<IDawrinCompatibleTarget>();
+
+    picker.placeholder = "Select a new target";
```

Comment:
I would probably move the QuickPick code outside of `commands.ts`. It would be good to generalise it if possible. I have another task #369 that'll need one. Maybe add a generalised version inside the UI folder. Provide list of menu entries which include a label and a function to call when selected (something like that).

Reply:
Good call, thanks! I extracted most of this into a generalized `withQuickPick` that handles the boilerplate.